### DISCUSSION TITLE
Bar with dashboard variables should be always visible

### DISCUSSION THREAD

**Discussion by ungolyant** [grafana/grafana#46088] [Dashboards, Panels & Visualizations]
**Steps to reproduce**:
Create or open a dashboard that has at least one dashboard variable shown (not hidden), and has many panels so scrollbar is shown on the right. Scroll the page down.

**Actual Result**:
Bar with variables will not be visible any more.

**Expected Result**
I would like (at least as an option in Dashboard settings page) that bar with dashboard variables is always visible. Reasons:
1. A variable value selected by the user can influence greatly on data shown (e.g. filtered) on visible panels/charts, thus data shown on a panel can be very misleading if user is not aware of current variable value.
2. There are use-cases when (often) changing of variable value is very useful, and scrolling up and down, as well as not being able to see immediate change of data shown on a panel can very frustrating.

**Environment**:
- Grafana version: v8.3.4 (a551d74b11)
- User OS & Browser: Windows 10, Firefox & Chrome

**Comment by ashharrison90** [comment]
hey @ungolyant 👋 

thanks for raising this!
agreed, we can make the experience better here. it's tricky though because there's also situations (e.g. if you have 20 or so variables that take up a lot of vertical space) where you wouldn't want that bar to stickily position to the top.

we're using github discussions to track feature requests so i'll transfer this to a discussion for now. please feel free to add upvotes there and when we're ready to commit to implementing it we'll convert it back to an issue.

hope that makes sense! 😄

**Comment by ungolyant** [comment]
Hello Ashley,
Thanks for the reply. I would say that the best option is to have a setting/checkbox in Dashboard Settings, so the dashboard creator can chose for itself weather he would like "pinned" parameters line or not. Or maybe in Dashboard Settings > Variables screen?

**Comment by ungolyant** [comment]
I am trying to mitigate this UX issue by splitting charts into many Dashboard Rows so I can collapse many Rows that are above a chart that I am looking at, but when I have too many Rows it is still a problem to see all of them plus toolbar with Dashboard Parameters...
![image](https://user-images.githubusercontent.com/1949199/171588382-0c5c701d-9662-43cc-87ce-7b5dd9206b32.png)
![image](https://user-images.githubusercontent.com/1949199/171588931-b50283c2-e1aa-49bc-8c59-567ab70d1098.png)

**Comment by fionera** [comment]
I would really like to see this as a feature. Even when you have them folded its very annoying to always scroll up and remove your focus away from the panels

**Comment by vasiliev123** [comment]
I would also love to see this implemented like @fionera said already. Many times we've got filters for servers and we need to move away our focus from panels to just change the server ;)
I'm also cheering for implementing an option in dashboard settings whether someone wants to see filters always on screen or not.

**Comment by patsevanton** [comment]
A very necessary feature have a setting/checkbox in Dashboard Settings

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This is being addressed in #11166
